### PR TITLE
Fixed atx processing

### PR DESCRIPTION
**Pull request by gavraz**
* Fixed proc ATX not storing id
* Strict non-genesis flow for app_test

### CODE REVIEW

**Review by antonlerner — APPROVED** [review]

File: activation/atxdb.go [hunk 1]
```diff
@@ -65,6 +65,11 @@ func (db *ActivationDb) ProcessAtx(atx *types.ActivationTx) {
 	if err != nil {
 		db.log.Error("cannot store atx: %v", atx)
 	}
+
+	err = db.ids.StoreNodeIdentity(atx.NodeId)
+	if err != nil {
+		db.log.Error("cannot store node identity: %v", atx.NodeId)
```
Comment (antonlerner) [inline activation/atxdb.go]: please include error message in print as well

Comment (gavraz) [reply activation/atxdb.go]: fixed

File: signing/signer.go [hunk 2]
```diff
@@ -20,6 +20,10 @@ func (p *PublicKey) Bytes() []byte {
 	return p.pub
 }
 
+func (p *PublicKey) Pub() string {
+	return string(p.pub)
```
Comment (antonlerner) [inline signing/signer.go]: please use some formatting

File: hare/integration_test.go [hunk 3]
```diff
@@ -32,8 +32,8 @@ type hareIntegrationThreeNodes struct {
 }
 
 func Test_16Nodes_HareIntegrationSuite(t *testing.T) {
+	t.Skip()
```
Comment (antonlerner) [inline hare/integration_test.go:35]: revert this

File: hare/integration_test.go [hunk 4]
```diff
@@ -82,8 +82,8 @@ type hareIntegration20Nodes struct {
 }
 
 func Test_20Nodes_HareIntegrationSuite(t *testing.T) {
+	t.Skip()
```
Comment (antonlerner) [inline hare/integration_test.go:84]: revert this

File: miner/block_builder.go [hunk 5]
```diff
@@ -242,13 +242,15 @@ func (t *BlockBuilder) acceptBlockData() {
 			return
 
 		case id := <-t.beginRoundEvent:
+			t.Info("Got begin round event id=%v", id)
```
Comment (antonlerner) [inline miner/block_builder.go]: is this print necessary?

File: cmd/node/node.go [hunk 6]
```diff
@@ -511,7 +512,8 @@ func (app *SpacemeshApp) Start(cmd *cobra.Command, args []string) {
 
 	app.unregisterOracle = func() { oracleClient.Unregister(true, app.edSgn.PublicKey().String()) }
 
-	rng := BLS381.DefaultSeed()
+	rng := amcl.NewRAND()
+	rng.Seed(10, app.edSgn.Sign([]byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}))
```
Comment (antonlerner) [inline cmd/node/node.go]: add comment, consider having a different byte array to sign (maybe the bls pub key? or any other pub key)

File: signing/signer.go (hunk 2 above)
Comment (gavraz) [reply signing/signer.go]: Removed

File: miner/block_builder.go (hunk 5 above)
Comment (gavraz) [reply miner/block_builder.go]: Removed

File: hare/integration_test.go (hunk 4 above)
Comment (gavraz) [reply hare/integration_test.go:84]: Skip for now (not used anyway for testing)

File: cmd/node/node.go (hunk 6 above)
Comment (gavraz) [reply cmd/node/node.go]: Fixed

**Pull request merged**